modules/Header/ui/CategoriesNav/CategoriesNavDesktop.tsx

```diff
@@ -34,6 +34,19 @@ export function CategoriesNavDesktop({
         (translatedCategory) => !getCategory(translatedCategory)?.is_featured,
     );
 
+    const [currentWidth, setCurrentWidth] = useState(0);
```

Comment:
We usually prefer to declare our `useState` calls on top of the function body.

```tsx
export function CategoriesNavDesktop({
    categories,
    translatedCategories,
    localeCode,
    buttonClassName,
    navigationItemClassName,
}: CategoriesNavDesktop.Props) {
    const [currentWidth, setCurrentWidth] = useState(0);
    // ...
```

Meanwhile, I'm not sure about the `currentWidth` as the variable name, how about we call it `documentWidth` to be extra clear? Because that's what it really is. Right?